### PR TITLE
[AAP-80579] Fix dynaconf 3.3.0 compat, rename in-files CI to canary

### PR DESCRIPTION
**Pull request by AlanCoding**
## Description

- **What is being changed?** Three things: (1) compatibility helpers for dynaconf 3.3.0's relocated internal attributes, (2) the CI `in-files` jobs are renamed to "Canary (unpinned deps, may fail)" and moved into their own job, (3) the canary jobs are restricted to run only on `devel` branch of `ansible/django-ansible-base`.
- **Why is this change needed?** dynaconf 3.3.0 moved `_post_hooks` and `_loaded_files` from `Settings` to `Settings.__core__.config` (dynaconf/dynaconf#1335). The `in-files` tox environments install from `.in` files (version ranges) rather than pinned `.txt`, so they picked up 3.3.0 and started failing. The canary job name also didn't communicate that failures are expected.
- **How does this change address the issue?** Adds `_dynaconf_post_hooks()` / `_dynaconf_loaded_files()` helpers that detect 3.3.0+ via `hasattr(settings, '__core__')` and route to the correct attribute location. Works with both dynaconf 3.2.x and 3.3.0. The CI restructuring makes the canary purpose clear and avoids noise on non-devel branches and forks.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Test update
- [ ] Refactoring (no functional changes)
- [x] Development environment change
- [ ] Configuration change

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [x] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [x] I have considered the security impact of these changes
- [x] I have considered performance implications
- [x] I have thought about error handling and edge cases
- [x] I have tested the changes in my local environment

### Prerequisites

- Python 3.11+ with a virtualenv

### Steps to Test

1. Install dynaconf 3.3.0: `pip install 'dynaconf==3.3.0'`
2. Run the affected test: `pytest test_app/tests/lib/dynamic_config/test_dynaconf_settings.py::test_load_python_file_with_injected_scope -x`
3. Downgrade to 3.2.12 and re-run: `pip install 'dynaconf==3.2.12'` then repeat step 2

### Expected Results

- Test passes on both dynaconf 3.2.12 and 3.3.0
- CI canary jobs only appear on PRs targeting `devel` in `ansible/django-ansible-base`

### Required Actions

- [ ] Requires documentation updates
- [ ] Requires downstream repository changes
- [ ] Requires infrastructure/deployment changes
- [ ] Requires coordination with other teams
- [ ] Blocked by PR/MR: #XXX

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved compatibility with different configuration library versions, helping settings files load more reliably across environments.
  * Updated file-loading tracking to avoid relying on internal details, reducing the chance of breakage during upgrades.

* **Chores**
  * Adjusted the CI pipeline to split out dependency-check runs into a separate job for the main development branch.
  * Streamlined the primary test matrix to focus on the standard automated checks.

### CODE REVIEW

**Comment by bmclaughlin** [comment]
PR #1034 Test Results

All tests pass on both dynaconf versions.

Code changes look correct:
- The hasattr(settings, '__core__') detection cleanly routes to the right attribute location on each version
- pragma: no cover on the 3.3.0 branches is appropriate since CI coverage runs pin dynaconf 3.2.12
- The CI workflow correctly separates canary (unpinned-deps) jobs into their own canary-deps job, gated to only run on devel branch of ansible/django-ansible-base

**Review by bmclaughlin — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -113,3 +105,46 @@ jobs:
...
+      matrix:
+        tests:
+          - env: py311-in-files
+            python-version: "3.11"
+          - env: py312-in-files
+            python-version: "3.12"
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          show-progress: false
```
[the remark this answers is not in the document]
Comment (bhavenst) [reply .github/workflows/ci.yml:132]: Good to do, but need to merge..  Can be done later..

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -113,3 +105,46 @@ jobs:
...
+
+      - name: Install build requirements
+        run: sudo apt-get update && sudo apt-get install -y libsasl2-dev libldap2-dev libssl-dev libxmlsec1-dev
+
+      - name: Install python ${{ matrix.tests.python-version }}
+        uses: actions/setup-python@v4
```
[the remark this answers is not in the document]
Comment (bhavenst) [reply .github/workflows/ci.yml:138]: Good to do, but need to merge..  Can be done later..
